app/lib/actions/katello/repository/fetch_pxe_files.rb

```diff
@@ -19,16 +19,19 @@ def run
                       else
                         SmartProxy.default_capsule!
                       end
-            repo_path = repository.full_path(capsule, true).chomp("/")
 
             os = Redhat.find_or_create_operating_system(repository)
+            medium_provider = Katello::ManagedContentMediumProvider.new(
+              OpenStruct.new(
+                content_source: capsule,
+                kickstart_repository: repository))
 
             ueber_cert = ::Cert::Certs.ueber_cert(repository.organization)
             cert = OpenSSL::X509::Certificate.new(ueber_cert[:cert])
             key = OpenSSL::PKey::RSA.new(ueber_cert[:key])
 
-            Redhat::PXEFILES.each_key do |pxe_file|
-              fetch("#{repo_path}/#{os.url_for_boot(pxe_file)}", cert, key)
+            os.boot_file_sources(medium_provider).values.each do |source_pxe_file|
+              fetch(source_pxe_file, cert, key)
             end
```

Comment:
@ShimShtein this works great!  Thanks!!! Would you be willing to move some of this into the repository model and write a test for it, so that we can catch this breakage in the future?